### ISSUE TITLE
Contact Card

### ISSUE THREAD

**Issue by chiwaileung** [MSFREACH/MSF-REACH-COLLABORATION#163]
[Contact Card feature – FEEDBACK.docx](https://github.com/MSFREACH/MSF-REACH-COLLABORATION/files/4266716/Contact.Card.feature.FEEDBACK.docx)
[Contact Card feature  FEEDBACK_Edited_commented.docx](https://github.com/MSFREACH/MSF-REACH-COLLABORATION/files/4266717/Contact.Card.feature.FEEDBACK_Edited_commented.docx)

From Requirement Document:
- [ ] Viewer or Public (with no MSF email address) should be able to consult and edit their contact details by 1) entering their email address 2) system check if they are in the database 3) they receive a link via email to go and retrieve their data to consult or edit - see old current production env.
- [x] Viewers should not be able to edit contact card - we remove the edit button for Viewers (only)
- [X]  Confidentiality consent to have from the contacts that you are entering information for/or for your contact details
- [x] Separating between individuals and organisation
- Field of information
  - organisation
    - [x] Organisation Name
    - [x] Activities
    - [x] Area of Activities
    - [x] Contact Person
      - [x] Phone Number
      - [x] Email Address
      - [x] Physical Address
  - individuals
    - [x] All those fields of information are not mandatory except 
      - [x] name
      - [x] area of activities
      - [x] email of contact person
- [x] For suppliers, adding a field of information that requests ‘Which services’ (reply as free text)
- [x] Adding a field of information named ‘Domain of Interest’ with dropdown menu including
  - [x] Migration
  - [x] Other (other + specify)
- [x] Adding a field of information named ‘Profile’ with dropdown menu including:
  - [x] Medical
  - [x] Non-Medical
  - [x] N/A

 - [x] Contact cards need to identify if users would like to enter a contact card or an organisation.
- [x] Two contact cards (organisation + contact) need to be available within REACH. a/contact stays as currently. 
- [x] organisation info card needs to request the following information: organisation name; activities; area of activities; contact person (phone number, email, address, etc.). All those fields of information are not mandatory except [name], [area of activities], [email of contact person]. 
- [x] Making contact field editable by operators 
- [x] Being able to enter no-geo located contact
- [ ] Being able to enter contact with no email address. 
- [x] the pin on the map are not user-friendly (too big at a certain level of scale). 
- [x] From the contact card, user is not sure that address is well added after the ‘map’ slide confirmed in the card itself. pin on the map is not user-friendly while filling up the form.
- [ ] ‘contact to share’ feature: interface needs to be reviewed for more intuitiveness. 
- [x] Add a field ‘which services’ in the contact card [adding a new contact], if the person ticks the box supplier in the [category] field of information.
- [x] Being able to upload list of contact through an excel file (rules to defined to be GDPR compliant).
- [ ] Display of organisation icon on landing page: users need to be able to see the area of activities by a box and differentiate the organisation from the person contact cards.
- [x] NGO should be a new marker on the menu selection on and off under ‘contacts. NGO and other actors mapping (areas of activities as general, same than responses, not being a pin) on landing page or per event?
- [ ] 	'Submit or update your contact details' after the report card is submitted - Wording of button to be changed.
- [x] Adding a field ‘domain of interest’ with dropdown menu (including ‘migration’, …). Dropdown still needs to be defined.
- [x] Adding a field ‘profile’ with a dropdown menu including three option: medical, non-medical, N/A

[Features clarification (contact card).docx](https://github.com/MSFREACH/MSF-REACH-COLLABORATION/files/4137462/Features.clarification.contact.card.docx)

**Comment by chiwaileung** [comment]
@hohangleung Please address the unchecked items in the list.

**Comment by chiwaileung** [comment]
@LucieGueuning 
There is a conflict with requirement
![image](https://user-images.githubusercontent.com/50646734/73996853-cc9dbb80-4997-11ea-89d4-4eed5256364a.png)
And
![image](https://user-images.githubusercontent.com/50646734/73996863-d7f0e700-4997-11ea-8ffa-ece717a50a4e.png)

**Comment by chiwaileung** [comment]
@LucieGueuning 
![image](https://user-images.githubusercontent.com/50646734/73997000-40d85f00-4998-11ea-999e-f0cbb3506f2b.png)
If the organisation is showing the polygon of area in the landing pages. That is not good for presentation. Imagine that, there will be many polygon overlapped and the user will not able to click on that. I suggest to use the point for the contact person on landing page, after user click on the point, it will show the polygon of activity area for the organisation. 
The same problems will be occurs for events also, please consider.

@hohangleung Please comments on my suggestion if you have any further suggestion.

**Comment by hohangleung** [comment]
> @LucieGueuning
> ![image](https://user-images.githubusercontent.com/50646734/73997000-40d85f00-4998-11ea-999e-f0cbb3506f2b.png)
> If the organisation is showing the polygon of area in the landing pages. That is not good for presentation. Imagine that, there will be many polygon overlapped and the user will not able to click on that. I suggest to use the point for the contact person on landing page, after user click on the point, it will show the polygon of activity area for the organisation.
> The same problems will be occurs for events also, please consider.
> 
> @hohangleung Please comments on my suggestion if you have any further suggestion.


Just to clarify a bit:
I supposed the 1st polygon will get removed when you click on another organisation?

It seems to be OK, however I suggest to use the center point of organisation polygon instead of another point from contact person, it's simpler.

**Comment by LucieGueuning** [comment]
> @LucieGueuning
> There is a conflict with requirement
> ![image](https://user-images.githubusercontent.com/50646734/73996853-cc9dbb80-4997-11ea-89d4-4eed5256364a.png)
> And
> ![image](https://user-images.githubusercontent.com/50646734/73996863-d7f0e700-4997-11ea-8ffa-ece717a50a4e.png)

The mandatory field is for organisation 
The requirement below is for contact as an individual

BUTI don't know who we will enter a contact without email address as it is the unique ID for now and the reference for the notification (which is nice to have feature) as well as the reminder system for keeping the database updated. Any idea?

**Comment by LucieGueuning** [comment]
> @LucieGueuning
> ![image](https://user-images.githubusercontent.com/50646734/73997000-40d85f00-4998-11ea-999e-f0cbb3506f2b.png)
> If the organisation is showing the polygon of area in the landing pages. That is not good for presentation. Imagine that, there will be many polygon overlapped and the user will not able to click on that. I suggest to use the point for the contact person on landing page, after user click on the point, it will show the polygon of activity area for the organisation.
> The same problems will be occurs for events also, please consider.
> 
> @hohangleung Please comments on my suggestion if you have any further suggestion.


Yes, you are right. 
event should be pin (refer to MSF responses polygone epicenter) - one pin per one OC responding (there is 5 OCs in MSF) 

BUT for contact is not working as individual is different from organisation. So how to differentiate the individual from the organisation from an iconography perspective on landing page? Any idea?

**Comment by LucieGueuning** [comment]
> > @LucieGueuning
> > ![image](https://user-images.githubusercontent.com/50646734/73997000-40d85f00-4998-11ea-999e-f0cbb3506f2b.png)
> > If the organisation is showing the polygon of area in the landing pages. That is not good for presentation. Imagine that, there will be many polygon overlapped and the user will not able to click on that. I suggest to use the point for the contact person on landing page, after user click on the point, it will show the polygon of activity area for the organisation.
> > The same problems will be occurs for events also, please consider.
> > @hohangleung Please comments on my suggestion if you have any further suggestion.
> 
> Just to clarify a bit:
> I supposed the 1st polygon will get removed when you click on another organisation?
> 
> It seems to be OK, however I suggest to use the center point of organisation polygon instead of another point from contact person, it's simpler.


I agree. But can you clarify icons and color for the following four : 
- individual - from MSF 
- Individual - not from MSF 
- organisation - epicenter of acitivities (obv not from MSF)

**Comment by hohangleung** [comment]
@LucieGueuning 
 Being able to enter contact with no email address.
email address is mandatory now, for cross check duplicate records

**Comment by LucieGueuning** [comment]
> @LucieGueuning
> Being able to enter contact with no email address.
> email address is mandatory now, for cross check duplicate records

Email address is mandatory for duplication records, for follow-up via email with contacts. BUT I have users that have contacts to be entered and do not have email address. They don't have computer as there are in the middle of nowhere. Do you have a suggestion? a solution? 

They have a phone number to be contacted.

**Comment by LucieGueuning** [comment]
Solution to be discussed with @msfallencheng to ensure we don't have issue with IT policy. 

Solution of creating fake email address in the back-end for non-email addressed contacts (not being users).

**Comment by LucieGueuning** [comment]
Solution: 
**If a contact (i.e. public - external to MSF; using the contact card standalone) is not entering an email address, then a dummy email address will be created in the database for identification purposes.** 
- [ ] In order to update the contact details, the admin of the right sub-group (see location) will be responsible to keep those people contact details updated (by calling them, etc.) - email will be sent for review on the same schedule than data retention one. 
- [ ] For data retention, those dummy email addresses will be out of the automation process of looking at contact active or not active - admin edit the contact that will trigger the activity, then trigger the data retention process. 
- [ ] Need to add a sentence next to email address to advice users to enter it for various reasons. 'It is recommended to provide your email address to be the owner and controller of your contact details. Please enter your email address hereafter.'
